### PR TITLE
Feature: Login API / User State / Auth State / Protected Route / Autologoff

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_API_HOST=/api/v1
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply .env:1]: Done

File: src/app/InactivityTracker.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+export type InactivityTrackerProps = PropsWithChildren
+
+const events = ["load", "click", "scroll", "keypress"]
+
+const ONE_SEC = 1000
+const ONE_MIN = 60 * ONE_SEC
+const LOGOUT_TIME_LIMIT = 15 * ONE_MIN // 15 min
+
+export const InactivityTracker = ({ children }: InactivityTrackerProps) => {
+  let timer: NodeJS.Timeout | undefined
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/InactivityTracker.tsx:18]: Done

File: src/app/InactivityTracker.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  const dispatch = useAppDispatch()
+  const navigate = useNavigate()
+
+  // this resets the timer if it exists.
+  const resetTimer = () => {
+    if (timer) clearTimeout(timer)
+  }
+
+  const logoutAction = () => {
+    localStorage.clear()
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/InactivityTracker.tsx:28]: Done

File: src/app/InactivityTracker.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  useState(() => {
+    Object.values(events).forEach(item => {
+      window.addEventListener(item, () => {
+        resetTimer()
+        logoutTimer()
+      })
+    })
+  })
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/InactivityTracker.tsx:56]: Done

File: src/app/routes.tsx [hunk 5]
```diff
@@ -1,14 +1,26 @@
...
+  const auth = useAppSelector(userAuthSelector)
+
   return (
     <Routes>
-      <Route path={ROUTES.main.path} element={<div>dashboard</div>} />
       <Route path={ROUTES.login.path} element={<LoginPage />} />
       <Route path={ROUTES.signup.path} element={<div>signup</div>} />
       <Route path={ROUTES.forgotPassword.path} element={<div>forgot password</div>} />
+
+      {auth?.token && (
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/routes.tsx:19]: Done

File: src/entities/user/model/auth.slice.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  name: "auth",
+  initialState,
+  reducers: {
+    setAuth: (state, action: PayloadAction<TAuthUserState>) => {
+      state = { ...state, ...action.payload }
+      return state
+    },
+    clearAuth: state => {
+      state = {}
+      return state
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/user/model/auth.slice.ts:24]: Done

File: src/entities/user/model/auth.slice.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  scope: [],
+}
+
+export const authUserSlice = createSlice({
+  name: "auth",
+  initialState,
+  reducers: {
+    setAuth: (state, action: PayloadAction<TAuthUserState>) => {
+      state = { ...state, ...action.payload }
+      return state
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/user/model/auth.slice.ts:20]: Done

File: src/entities/user/model/user.slice.ts [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  name: "user",
+  initialState,
+  reducers: {
+    setUser: (state, action: PayloadAction<UserState>) => {
+      state = { ...state, ...action.payload }
+      return state
+    },
+    clearUser: state => {
+      state = {}
+      return state
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/user/model/user.slice.ts:27]: Done

File: src/entities/user/model/interface.ts [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import * as z from "zod"
+import { AuthSchema, UserSchema, UserStateSchema } from "./user.schema"
+
+export type TUserSchema = z.infer<typeof UserSchema>
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/entities/user/model/interface.ts:4]: Done

File: src/pages/Login/lib/api.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  const response = await api.get("/user/authentication", { headers })
+  return response
+}
+
+export const useFetchAuthorization = (options: MutationOptions<ResponseData, ResponseError, TLoginForm>) => {
+  const name = "auth"
+
+  const fetcher = useMutation([name], (data: TLoginForm) => makeRequest(data), options)
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/pages/Login/lib/api.ts:26]: It is a common case. Work properly.

File: src/pages/Login/ui/index.tsx [hunk 11]
```diff
@@ -54,7 +73,11 @@ const LoginPage = () => {
             </Container>
 
             <Container>
-              <BasicButton type="submit" variant="primary">
+              <BasicButton
+                type="submit"
+                variant="primary"
+                disabled={!isObjectEmpty(errors) || mutation.isLoading}
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/pages/Login/ui/index.tsx:79]: Done

File: src/pages/Login/model/types.ts [hunk 12]
```diff
@@ -1,5 +1,43 @@
...
+    applets: Record<string, unknown>
+    isDefaultName: boolean
+  }
+  authToken: {
+    expires: string
+    scope: string[]
+    token: string
+  }
+  message: string
+  user: {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/pages/Login/model/types.ts]: Done

**Pull request merged**